### PR TITLE
AG-7097 - Improve AreaSeries stoke rendering to avoid overlapping with other series.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-7097

Alternative/improved approach vs. #5439 - improve rendering order control within a series.

Bonus:
- Moves entire stacked series `zIndex` in lock-step to avoid partial-stack rendering.
- Fixes `lineCap` and `lineJoin` typings.

### CODE REVIEW

File: charts-packages/ag-charts-community/src/util/compare.ts [hunk 1]
```diff
@@ -3,3 +3,37 @@ export type Comparator<T> = (a: T, b: T) => number;
...
+        zDiff = 1;
+    } else {
+        zDiff = String(a).localeCompare(String(b));
+    }
+
+    return zDiff;
+}
+
+export function compoundAscending<E, A extends Array<E>>(a: A, b: A, comparator: Comparator<E>): number {
+    for (const idx in a) {
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/util/compare.ts:30]: We want the index of each element as we need to lookup the parallel index from `b`, `for ... of` would require us to track the index ourselves, this is just neater IMO in this particular case.

File: charts-packages/ag-charts-community/src/util/compare.ts [hunk 2]
```diff
@@ -3,3 +3,37 @@ export type Comparator<T> = (a: T, b: T) => number;
 export function ascending<T>(a: T, b: T): number {
     return a < b ? -1 : a > b ? 1 : a >= b ? 0 : NaN;
 }
+
+export function ascendingStringNumberUndefined(
+    a: number | string | undefined | null,
+    b: number | string | undefined | null
+): number {
+    let zDiff = 0;
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/util/compare.ts]: Good spot, renamed.

File: charts-packages/ag-charts-community/src/chart/series/series.ts [hunk 3]
```diff
@@ -344,7 +347,7 @@ export abstract class Series<C extends SeriesNodeDataContext = SeriesNodeDataCon
 
         if (datum && highlightedDatum !== datum && itemId !== datum.itemId) {
             // Highlighting active, this series item not highlighted.
```
[the remark this answers is not in the document]
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/chart/series/series.ts]: Tweaked to be more accurate.

**Review by manapeirov — APPROVED** [review]
LGTM :)

**Pull request merged**